src/actions/copy-tpl-async.ts

```diff
@@ -1,40 +1,31 @@
-import fs from 'fs/promises';
-import { isBinary, processTpl } from '../util.js';
+import { isBinary } from '../util.js';
 import type { MemFsEditor } from '../index.js';
 import ejs from 'ejs';
-import type { CopyAsyncOptions } from './copy-async.js';
 
 export default async function (
   this: MemFsEditor,
   from: string | string[],
   to: string,
   data: ejs.Data = {},
   tplOptions?: ejs.Options,
-  options?: CopyAsyncOptions,
+  options?: Parameters<MemFsEditor['copyAsync']>[2],
 ) {
-  await this.copyAsync(
-    from,
-    to,
-    {
-      processDestinationPath: (path) => path.replace(/.ejs$/, ''),
-      ...options,
-      processFile(filename) {
-        if (isBinary(filename)) {
-          return fs.readFile(filename);
-        }
+  await this.copyAsync(from, to, {
+    ...options,
+    async fileTransform(destPath: string, sourcePath: string, contents: Buffer): Promise<[string, string | Buffer]> {
+      const processedPath = await ejs.render(destPath, data, { ...tplOptions, async: true });
```

Comment:
EJS async support is unusual.
Sync templates may not be compatible with async when using include.
```suggestion
      const processedPath = await ejs.render(destPath, data, tplOptions);
```

Reply:
Yeah, I was surprised by that too! I kept the path template async because there's no `filename` attached, so we know there'll be no `include()` in the template. That being said, I don't know if there's other tricky behaviors... I'm okay keeping them all sync.